OrbitCore/CallstackData.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_CALLSTACK_DATA_H_
+#define ORBIT_CORE_CALLSTACK_DATA_H_
+
+#include <memory>
+
+#include "BlockChain.h"
+#include "Callstack.h"
+#include "CallstackTypes.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
+
+class CallstackData {
+ public:
+  explicit CallstackData() = default;
+
+  void AddCallstackEvent(orbit_client_protos::CallstackEvent&& callstack_event);
+  void AddUniqueCallStack(CallStack call_stack);
+  void AddCallStackFromCallstackData(
+      const orbit_client_protos::CallstackEvent& event,
+      const CallstackData& other);
+
+  [[nodiscard]] const BlockChain<orbit_client_protos::CallstackEvent,
+                                 16 * 1024>&
+  GetCallstackEvents() const { return callstack_events_; };
+
+  [[nodiscard]] uint32_t GetCallstackEventsSize() const {
+    return callstack_events_.size();
+  };
+
+  const CallStack& GetCallStack(CallstackID callstack_id) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return *unique_callstacks_.at(callstack_id);
```

Comment:
This will fail, if the callstack is not available.
You should either add a comment, or maybe even better, consider using `find` and return a bare pointer or an `outcome`.

Reply:
Done